src/views/Timesheets/components/SavedShiftsList/index.js

```diff
@@ -83,6 +84,21 @@ export default class SavedShiftsList extends Component {
     });
   }
 
+  componentDidMount() {
+    let {shifts, directSupervisor, reportingSupervisor} = this.props;
+    let supervisorIdsReady = directSupervisor && reportingSupervisor;
+    let shouldGetSupervisors; 
+    if (shifts.length > 0) {
+      shouldGetSupervisors = this.props.cardTitle === "Saved Shifts" && (shifts[0].EML !== this.prevJob || (!this.state.directSupervisor || !this.state.reportingSupervisor));
+      this.prevJob = shifts[0].EML
+    } else {
+      shouldGetSupervisors = this.props.cardTitle === "Saved Shifts" && supervisorIdsReady && (!this.state.directSupervisor || !this.state.reportingSupervisor) && shifts !== null;
+    }
+    if (shouldGetSupervisors) {
+      this.getSupervisors();
+    }
+  }
+
   componentDidUpdate() {
     let {shifts, directSupervisor, reportingSupervisor} = this.props;
```

Comment:
I think adding a little description of what exactly is happening here would be good.